bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/Workbench.java

```diff
@@ -1483,6 +1489,93 @@ IWorkbenchWindow createWorkbenchWindow(IAdaptable input, IPerspectiveDescriptor
 		return result;
 	}
 
+	private void placeNearActiveShell(MWindow window) {
+		if (getDisplay() == null) {
+			return;
+		}
+
+		Shell activeShell = getDisplay().getActiveShell();
+		if (activeShell == null) {
+			return;
+		}
+
+		Monitor currentMonitor = findMonitor(activeShell.getBounds());
+
+		final int padding = 20;
+		Rectangle paddedMonitorBounds = shrink(currentMonitor.getBounds(), padding);
+
+		final int offsetToExistingShell = 100;
+		Rectangle newShellBounds = new Rectangle(activeShell.getBounds().x + offsetToExistingShell,
+				activeShell.getBounds().y + offsetToExistingShell, window.getWidth(), window.getHeight());
+
+		moveIntoBounds(newShellBounds, paddedMonitorBounds);
+
+		window.setX(newShellBounds.x);
+		window.setY(newShellBounds.y);
+	}
+
+	private static Rectangle shrink(Rectangle rectangle, int padding) {
+		return new Rectangle(rectangle.x + padding, rectangle.y + padding, rectangle.width - 2 * padding,
+				rectangle.height - 2 * padding);
+	}
+
+	private Monitor findMonitor(Rectangle r) {
```

Comment:
The code became harder to read because of this non-speaking method name. I would actually propose to integrate this functionality as fallback behavior into `findMonitorThatContainsMostOf`. You could simply initialize the monitor variable with the primary monitor, which makes the code more concise and still valid (in case the rectangle is not part of any monitor, the primary monitor is one that contains most of the rectangle).

Reply:
👍 
Done in f15c036f30e9a2f990fd36b4dbc610d33cebae28